.eslintrc.js

```diff
@@ -77,203 +77,176 @@ const restrictedImportPatterns = [
-    plugins: ['react-native-a11y'],
-    parser: 'babel-eslint',
-    ignorePatterns: ['!.*', 'src/vendor', '.github/actions/**/index.js', 'desktop/dist/*.js', 'dist/*.js', 'node_modules/.bin/**', 'node_modules/.cache/**', '.git/**'],
+    extends: [
+        'expensify',
+        'airbnb-typescript',
+        'plugin:storybook/recommended',
+        'plugin:react-native-a11y/basic',
+        'plugin:@dword-design/import-alias/recommended',
+        'plugin:@typescript-eslint/recommended-type-checked',
+        'plugin:@typescript-eslint/stylistic-type-checked',
+        'plugin:you-dont-need-lodash-underscore/all',
+        'prettier',
+    ],
+    plugins: ['@typescript-eslint', 'jsdoc', 'you-dont-need-lodash-underscore', 'react-native-a11y', 'react'],
+    parser: '@typescript-eslint/parser',
+    parserOptions: {
+        project: './tsconfig.json',
+    },
     env: {
         jest: true,
     },
     globals: {
         __DEV__: 'readonly',
     },
-    overrides: [
-        {
-            files: ['*.js', '*.jsx', '*.ts', '*.tsx'],
-            plugins: ['react'],
-            rules: {
-                'prefer-regex-literals': 'off',
-                'rulesdir/no-multiple-onyx-in-file': 'off',
-                'react-native-a11y/has-accessibility-hint': ['off'],
-                'react/jsx-no-constructed-context-values': 'error',
-                'react-native-a11y/has-valid-accessibility-descriptors': [
-                    'error',
-                    {
-                        touchables: ['PressableWithoutFeedback', 'PressableWithFeedback'],
-                    },
-                ],
-                '@dword-design/import-alias/prefer-alias': [
-                    'warn',
-                    {
-                        alias: {
-                            '@assets': './assets',
-                            '@components': './src/components',
-                            '@hooks': './src/hooks',
-                            // This is needed up here, if not @libs/actions would take the priority
-                            '@userActions': './src/libs/actions',
-                            '@libs': './src/libs',
-                            '@navigation': './src/libs/Navigation',
-                            '@pages': './src/pages',
-                            '@styles': './src/styles',
-                            // This path is provide alias for files like `ONYXKEYS` and `CONST`.
-                            '@src': './src',
-                            '@desktop': './desktop',
-                            '@github': './.github',
-                        },
-                    },
-                ],
+    rules: {
+        // TODO: Remove the following rules after TypeScript migration is complete.
+        '@typescript-eslint/no-unsafe-argument': 'off',
+        '@typescript-eslint/no-unsafe-call': 'off',
+        '@typescript-eslint/no-unsafe-member-access': 'off',
+        '@typescript-eslint/no-unsafe-assignment': 'off',
```

Comment:
Do we still need these ones?

Reply:
I checked it yesterday. When I remove it I get around 1000 new TS errors, I'm going to handle these in a separate PR 😮‍💨

Reply:
🙈